util/template/expression_template.go

```diff
@@ -40,6 +40,17 @@ func expressionReplace(w io.Writer, expression string, env map[string]interface{
 		return fmt.Fprintf(w, "{{%s%s}}", kindExpression, expression)
 	}
 
+	if (hasVariableInExpression(unmarshalledExpression, "lastRetry.exitCode") && !hasVarInEnv(env, "lastRetry.exitCode") ||
```

Comment:
Oh, this hack is horrid. But it mirrors what's happening with retries, so I guess we go with it for now.

Can we make this less evil code wise - perhaps list of all the variables we're going to do this for and iterate? Including retry?

Reply:
Sure I think we can  do this for lastRetry but I do not want to do this for retries. I think they need to be distinct.
I could maybe be refactored to share some common logic though.